### PR TITLE
Added ability to select arbitrary scattering frequency and temperature

### PR DESCRIPTION
**Pull request by josephhardinee**
This adds the ability to use arbitrary temperatures and frequencies. Nick, can you take a look at this to make sure it still works for you. This changes the API for calling a few functions, but they should be a little bit easier to use now.

### CODE REVIEW

File: pydisdrometer/DropSizeDistribution.py [hunk 1]
```diff
@@ -133,30 +131,29 @@ def __init__(self, reader, time_start = None, location=None,
...
         self.numt = len(reader.time['data'])
         location = {}
 
         if location:
             self.location = {'latitude': location[0], 'longitude': location[1]}
 
-    def change_scattering_temperature(self, scattering_temp='10C'):
+        self.set_scattering_temperature_and_frequency()
+
+    def set_scattering_temperature_and_frequency(self, scattering_temp=10, scattering_frequency = 9.7e9):
```
Comment (nguy) [inline pydisdrometer/DropSizeDistribution.py]: I'd stay consistent with the naming conventions either ```scattering_temp``` and ```scattering_freq``` or ```scattering_temperature``` and ```scattering_frequency```

File: pydisdrometer/DropSizeDistribution.py [hunk 2]
```diff
@@ -133,30 +131,29 @@ def __init__(self, reader, time_start = None, location=None,
...
         ''' Change the scattering temperature. After use, re-run calculate_radar_parameters
-        to see the effect this has on the parameters
+        to see the effect this has on the parameters. Temperatures are in Celsius. Defaults to 10C X-band.
 
         Parameters
         ----------
-        temp: optional, string
+        temp: optional, float
```
Comment (nguy) [inline pydisdrometer/DropSizeDistribution.py]: Should change the docstring to indicate the new float. 
Something like: "Scattering temperature [C]"?

File: pydisdrometer/DropSizeDistribution.py [hunk 3]
```diff
@@ -133,30 +131,29 @@ def __init__(self, reader, time_start = None, location=None,
...
             String of temperature to scatter at. Choice of ("0C","10C","20C").
```
Comment (nguy) [inline pydisdrometer/DropSizeDistribution.py]: Should change the docstring to indicate the new float.
Something like: "Scattering temperature [C]"?

Comment (nguy) [inline pydisdrometer/DropSizeDistribution.py]: Also, these parameters are no longer correct.

**Comment by nguy** [comment]
Also, probably should roll the rev version for this.

File: pydisdrometer/utility/dielectric.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    """
+
+    es = s[0] + s[1] * temp + s[2] * temp**2 + s[3] * temp**3
+    ep = es - (2 * np.pi * freq)**2 * (A_i(0, temp, freq) + A_i(1, temp, freq))
+    epp =  2 * np.pi * freq * (B_i(0, temp, freq) + B_i(1, temp, freq))
+
+    return np.sqrt(ep + 1j * epp)
+
+
+def A_i(i, temp, freq ):
```
Comment (nguy) [inline pydisdrometer/utility/dielectric.py]: remove space before ending paranthesis

**Comment by nguy** [comment]
@josephhardinee I pushed some modifications to this branch. I took care of most of the comments I made, see if you agree with them.

**Comment by nguy** [comment]
I don't follow how [these lines](https://github.com/josephhardinee/PyDisdrometer/blob/dielectric_calculation/pydisdrometer/DropSizeDistribution.py#L197-200) work. The Nd array is a 2D array and I get an error here. 

Note to us here that the Nd arrays in the ```Image2DReader``` need to reverse dimensions.

**Comment by josephhardinee** [comment]
@nguy: For those lines I'm just stepping through each time index and forming a DSD from each measured DSD.

**Comment by nguy** [comment]
Okay, well if you are okay with the mods I pushed, we could probably merge.

**Pull request merged**